### PR TITLE
[CHORE]: move uri functions to new "daft-functions" crate

### PR DESCRIPTION
**Pull request by universalmind303**
so i totally forgot about [typetag](https://github.com/dtolnay/typetag) when I did https://github.com/Eventual-Inc/Daft/pull/2466. So the whole `registry` thing isn't even needed anymore, which greatly simplifies things.

### CODE REVIEW

File: src/daft-dsl/Cargo.toml [hunk 1]
```diff
@@ -4,7 +4,6 @@ common-error = {path = "../common/error", default-features = false}
 common-io-config = {path = "../common/io-config", default-features = false}
 common-treenode = {path = "../common/treenode", default-features = false}
 daft-core = {path = "../daft-core", default-features = false}
-daft-io = {path = "../daft-io", default-features = false}
```
Comment (samster25) [inline src/daft-dsl/Cargo.toml:7]: Beautiful!

File: src/daft-plan/Cargo.toml [hunk 2]
```diff
@@ -8,6 +21,7 @@ common-py-serde = {path = "../common/py-serde", default-features = false}
 common-treenode = {path = "../common/treenode", default-features = false}
 daft-core = {path = "../daft-core", default-features = false}
 daft-dsl = {path = "../daft-dsl", default-features = false}
+daft-functions = {path = "../daft-functions", default-features = false}
```
Comment (samster25) [inline src/daft-plan/Cargo.toml:24]: we should only need this a dev dependency right? It looks like today we only use it in a test

**Review by samster25 — APPROVED** [review]
Wow this is much cleaner overall for inter-crate functions!

File: src/daft-plan/Cargo.toml (hunk 2 above)
Comment (universalmind303) [reply src/daft-plan/Cargo.toml:24]: good call! will update.

**Pull request merged**